Some improvements to the v11 migration guide (#762)
rescript-association/rescript-lang.org: pages/docs/manual/latest/migrate-to-v11.mdx

---
title: "Migrate to v11"

> description: "Instructions on upgrading to ReScript 11 with the new uncurried mode"

Reviewer: ```suggestion
description: "Instructions on upgrading to ReScript 11"
```

canonical: "/docs/manual/latest/migrate-to-v11"
---

# Migrate to ReScript 11

## Foreword

The ReScript community is proud to introduce ReScript V11 which comes with a ton of new features but also removes a lot of bulk.
A migration to it can be very straightforward, but it can also take some time, depending on your code style or what dependencies you use.

Please have a look at the full [set of breaking changes](#list-of-all-breaking-changes) below to be able to decide whether this is a task you want to undertake. There is also the possibilty to opt-out of uncurried mode for now, which is probably the most fundamental change of this release. That and other new and notable features are discussed in the following blogposts:

- [Better interop with customizable variants](/blog/improving-interop)
- [Enhanced Ergonomics for Record Types](/blog/enhanced-ergonomics-for-record-types)
- [First-class Dynamic Import Support](/blog/first-class-dynamic-import-support)
- [Uncurried Mode](/blog/uncurried-mode)

> ## Minimal Migration

Reviewer: I am not sure this should be at the most prominent place.
Can we move it to the bottom and link to it from the sentence with "opt-out of uncurried mode" above?

This guide describes the things to do at least to migrate to ReScript 11.

### Disable uncurried mode

If you use currying extensively and don't want to bother with adapting your code, or have dependencies that just don't work with uncurried mode yet, just set it to false in your `rescript.json`.

```json
{
  "uncurried": false
}
```

For more information, read the Build System Configuration about [uncurried](/docs/manual/latest/build-configuration#uncurried).

## Recommended Migration

### Uncurried Mode

For uncurried mode to take effect in ReScript 11 there is nothing to configure, it is activated by default.

### Adapt suffix

> ReScript 11 now allows having arbitrary suffixes in the generated JavaScript files. However it is still recommended to stick to using `.res.js`, `.res.mjs` or `.res.cjs`. For more information, read the Build System Configuration about [suffixes](/docs/manual/latest/build-configuration#suffix).

Reviewer: ```suggestion
ReScript 11 now allows having arbitrary suffixes in the generated JavaScript files. However, it is still recommended to stick to using `.res.js`, `.res.mjs` or `.res.cjs`. For more information, read the Build System Configuration about [suffixes](/docs/manual/latest/build-configuration#suffix).
```

### rescript.json

> The old configuration filename `bsconfig.json` is finally deprecated. Rename it to `rescript.json`.

Reviewer: ```suggestion
The old configuration filename `bsconfig.json` is finally deprecated. Rename `bsconfig.json` to `rescript.json` to get rid of the deprecation warning.
```

### ReScript Core standard library

> There is a new standard library in development that can already be installed called ReScript Core. It replaces the complete `Js` module as well as some of the more frequently used modules from `Belt` and is the recommended standard library to use with uncurried mode. At a later stage it will be integrated into the compiler while `Belt` will be maintaned as a separate npm package.

Reviewer: ```suggestion
[ReScript Core](https://github.com/rescript-association/rescript-core) is ReScript's new standard library. It replaces the complete `Js` module as well as some of the more frequently used modules from `Belt` and is the recommended standard library to use with uncurried mode.
```

> Until it's integrated in the compiler, it needs to be installed manually:

Reviewer: ```suggestion
It will be integrated into the compiler in a future version. In ReScript 11, it still needs to be installed manually:
```
